datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -56,15 +56,20 @@ public BookController(BookRepository repository, BookModelAssembler assembler, B
         }     
     }
 
+    private boolean amILeader(){
+        return this.storeRepository.findAll().get(0).getServerId().equals(this.leader.getLeader());
+    }
+
     @PostMapping("/bookstores/book")
     protected CollectionModel<EntityModel<Book>> oneBookToManyStores(@RequestBody Book book, @RequestParam List<String> id) throws Exception {
         List<EntityModel<Book>> entityList = new ArrayList<>();
-        if (!this.storeRepository.findAll().get(0).getServerId().equals(this.leader.getLeader())) {
+        if (!amILeader()) {
             String address = this.map.get(this.leader.getLeader());
-            address = address.substring(0,address.lastIndexOf("/") + 1) + "book?id=" + id.toString().replaceAll("[\\[ \\]]", "");
+            address = removeIDNum(address) + "book?id=" + String.join(",", id);
             Gson gson = new Gson();
             String json = gson.toJson(book.makeJson());
             HttpResponse<String> response = getPostConnectionJava9(address, json);
+            if(response.statusCode() != 200) throw new Exception();
```

Comment:
Im sure we can do better here :). 

how about `RuntimeException("Could not connect to....")`